src/Features/Lsif/Generator/ResultSetTracking/IResultSetTrackerExtensions.cs

```diff
@@ -17,7 +17,36 @@ internal static class IResultSetTrackerExtensions
         /// <summary>
         /// Returns the ID of the <see cref="ReferenceResult"/> for a <see cref="ResultSet"/>.
         /// </summary>
-        public static Id<ReferenceResult> GetResultSetReferenceResultId(this IResultSetTracker tracker, ISymbol symbol, IdFactory idFactory)
-            => tracker.GetResultIdForSymbol(symbol, Methods.TextDocumentReferencesName, () => new ReferenceResult(idFactory));
+        public static Id<ReferenceResult> GetResultSetReferenceResultId(this IResultSetTracker tracker, ISymbol symbol)
+            => tracker.GetResultIdForSymbol(symbol, Methods.TextDocumentReferencesName, static (_, idFactory) => new ReferenceResult(idFactory));
+
+        /// <summary>
+        /// Fetches the moniker node for a symbol; this should only be called on symbols where <see cref="SymbolMoniker.HasMoniker"/> returns true.
+        /// </summary>
+        public static Id<Moniker> GetMoniker(this IResultSetTracker tracker, ISymbol symbol, Compilation sourceCompilation)
+        {
+            return tracker.GetResultIdForSymbol(symbol, "moniker", (symbol, idFactory) =>
+            {
+                var moniker = SymbolMoniker.Create(symbol);
+
+                string? kind;
+
+                if (symbol.Kind == SymbolKind.Namespace)
+                {
+                    kind = null;
+                }
+                else if (symbol.ContainingAssembly.Equals(sourceCompilation.Assembly))
+                {
+                    kind = "export";
```

Comment:
what does 'export/import' mean here?

Reply:
https://microsoft.github.io/language-server-protocol/specifications/lsif/0.6.0/specification/#exportsImports

Basically when we're indexing a project, we will stick a moniker for symbols we encounter. 'export' means it came from this project (and is visible elsewhere), 'import' means it came from another project. I'm not sure if the rest of the consumers really use this much, but that's the spec at least.